torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -172,7 +172,7 @@ Reducer::Reducer(
       at::TensorOptions options;
       options = options.dtype(at::kInt);
       local_used_maps_[i] =
-          at::zeros({static_cast<long>(variable_count)}, options);
+          at::zeros({static_cast<long>(variable_count)}, options.pinned_memory(true));
```

Comment:
Hitting this in CI tests. How should we handle this properly? @pietern @mrshenli Thanks.
```
Nov 26 06:19:00   File "test_c10d.py", line 2937, in test_single_dtype_single_bucket
Nov 26 06:19:00     dist.Reducer([parameters], buckets, self.process_group)
Nov 26 06:19:00 RuntimeError: Pinned memory requires CUDA. PyTorch splits its backend into two shared libraries: a CPU library and a CUDA library; this error has occurred because you are trying to use some CUDA functionality, but the CUDA library has not been loaded by the dynamic linker for some reason.  The CUDA library MUST be loaded, EVEN IF you don't directly use any symbols from the CUDA library! One common culprit is a lack of -Wl,--no-as-needed in your link arguments; many dynamic linkers will delete dynamic library dependencies if you don't depend on any of their symbols.  You can check if this has occurred by using ldd on your binary to see if there is a dependency on *_cuda.so library. (getPinnedMemoryAllocator at ../aten/src/ATen/detail/CUDAHooksInterface.h:107)
```

Reply:
I think it will be fine if you make `options.pinned_memory(true)` conditional on `replicas_[i][0].is_cuda()`. The asynchronous copy becomes synchronous if you're copying from CPU tensor to CPU tensor, so you can keep the rest as is.

Reply:
Thanks. Done with 5ec6672. Originally I thought there is some global config for this purpose.